src/hotspot/cpu/x86/x86.ad

```diff
@@ -4487,6 +4513,46 @@ instruct vReplS_reg(vec dst, rRegI src) %{
   ins_pipe( pipe_slow );
 %}
 
+#ifdef _LP64
+instruct ReplHF_imm(vec dst, immH con, rRegI rtmp) %{
+  match(Set dst (Replicate con));
+  effect(TEMP rtmp);
+  format %{ "replicateHF $dst, $con \t! using $rtmp as TEMP" %}
+  ins_encode %{
+    int vlen_enc = vector_length_encoding(this);
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    assert(VM_Version::supports_avx512_fp16() && bt == T_SHORT, "");
+    __ movl($rtmp$$Register, $con$$constant);
+    __ evpbroadcastw($dst$$XMMRegister, $rtmp$$Register, vlen_enc);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct ReplHF_short_reg(vec dst, rRegI src) %{
+  predicate(VM_Version::supports_avx512_fp16() && Matcher::vector_element_basic_type(n) == T_SHORT);
+  match(Set dst (Replicate (ReinterpretS2HF src)));
+  format %{ "replicateHF $dst, $src" %}
+  ins_encode %{
+    int vlen_enc = vector_length_encoding(this);
+    __ evpbroadcastw($dst$$XMMRegister, $src$$Register, vlen_enc);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct ReplHF_reg(vec dst, regF src, rRegI rtmp) %{
```

Comment:
Hi @jatin-bhateja , there are JTREG tests testing Replicate with immediate FP16 values (`Rep1HF_imm` backend rule) but I noticed that there are no JTREG tests for testing these rules - `Rep1HF_reg` and `Rep1HF_short_reg`. Should those be added as well? Something like - 

```
float f = 10.0f;
for (int i = 0; i < SIZE: ++i) {
    res[i] = Float16.fma(a[i], b[i], Float16.valueOf(f));   // where f is a loop invariant float variable defined out of the loop 
}
```